contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -250,11 +173,101 @@ def _transcribe(item, parents, editRate, masterMobs):
+
+        def _find_mobslot_timecode(mob, phys_track_num):
+            timecode = (0, 0.0)
+            slot = [s for s in mob.slots if s['PhysicalTrackNumber'].value == phys_track_num and s.media_kind == 'Timecode']
+            tc = slot[0].segment if len(slot) > 0 else None
+            if isinstance(tc, aaf2.components.Sequence):
+                comps = [c for c in tc.components]
+                tc = comps[0] if len(comps) > 0 else None
+            if isinstance(tc, aaf2.components.Timecode):
+                timecode = (tc.start, slot[0].edit_rate.__float__())
+            return timecode
+
+        def _find_mob_chain_and_timecode(source_clip, editRate):
+            """ This is combining several processes in to one for efficieny.
+                 - Firstly, it traverses the SourceClip's Mob's structure using the
+                clip_id to return all the necessary Mobs.
+                 - Secondly, it extracts the necessary Start values for each object, and
+                converts the rate if necessary. This value can come from several different
+                places.
+                 - Lastly, it stores the individual SourceClips so that at the end of the
+                process the correct Length can be derived from the MasterMob"""
+
+            if not isinstance(source_clip, aaf2.components.SourceClip):
+                raise AAFAdapterError("Error: _find_mob_chain() requires a SourceClip component"
+                                      " - cannot parse {}".format(type(source_clip)))
+
+            mob_chain = [source_clip.mob]
+            source_clip_chain = []
+            timecode_chain = []
+            slot_id = source_clip.slot_id
+
+            for mob in mob_chain:
+                if hasattr(mob, 'descriptor') \
+                        and mob.descriptor.name in ['TapeDescriptor', 'ImportDescriptor']:
+                    timecode_chain.append(_find_mobslot_timecode(mob, 1))
+                    # Doing this so mob_chain and source_clip_chain will always have some
+                    # number of values for zip function below
+                    source_clip_chain.append(None)
+                elif mob:
+                    slot = [s for s in mob.slots if s.slot_id == slot_id]
+                    slot = slot[0] if len(slot) > 0 else None
+                    mob_source_clip = _find_source_clip(slot.segment) if slot else None
+                    if mob_source_clip and mob_source_clip.mob:
+                        slot_id = mob_source_clip.slot_id
+                        mob_chain.append(mob_source_clip.mob)
+                        source_clip_chain.append(mob_source_clip)
+                        timecode_chain.append(_find_mobslot_timecode(mob, slot['PhysicalTrackNumber'].value))
+                        timecode_chain.append((mob_source_clip.start, slot.edit_rate.__float__()))
+
+            frame_count = 0
```

Comment:
I haven't really used the RationalTime class at all. Is there an inherent advantage to using it for this process over dealing with the straight `frame_count`? I guess if it's doing the rate conversion automatically that's useful, but I'll need to do a bit of testing to make sure I'm getting the same results.

Reply:
The main reason I'd lean toward RationalTime is that it would allow you to mix rates and then in the end snap the value to a specific frame. This could prevent a lot of the rate checking you're having to do just below this.